src/ContentValidationListener.php

```diff
@@ -213,8 +213,20 @@ public function onRoute(MvcEvent $e)
         }
 
         $inputFilter->setData($data);
-        if ($inputFilter->isValid()) {
-            return;
+        /**
+         * A specific Filter may throw an Exception if an invalid value is given.
+         * Convert any Exception to return a failed validation response
+         */
+        try {
+            if ($inputFilter->isValid()) {
+                return;
+            }
+        } catch (FilterInvalidArgumentException $ex) {
```

Comment:
As noted by another reviewer, this is not a valid exception type.

In fact, looking through the various exceptions thrown by `Zend\Filter`, the only time I see `InvalidArgumentException` thrown is typically for setters; `DateTimeFormatter` and the various `File` filters are anomalies here.

Additionally, also noted by another reviewer, validators may also throw exceptions, so type-hinting against filter exceptions only will mean those bubble outwards.

As such, this should likely be hinting against `Exception`, which also means that the message must change. I'll note that below.